framework/src/st_things/things_stack/src/common/framework/things_data_manager.c

```diff
@@ -1260,23 +1263,72 @@ static int parse_things_info_json(const char *filename)
 				cJSON *certificate = cJSON_GetObjectItem(file_path, KEY_CONFIGURATION_FILEPATH_CERTIFICATE);
 				cJSON *privateKey = cJSON_GetObjectItem(file_path, KEY_CONFIGURATION_FILEPATH_PRIVATEKEY);
 
-				if (NULL != svrdb && NULL != provisioning && NULL != certificate && NULL != privateKey) {
-					memset(g_svrdb_file_path, 0, (size_t) MAX_FILE_PATH_LENGTH);
-					memset(g_certificate_file_path, 0, (size_t) MAX_FILE_PATH_LENGTH);
-					memset(g_private_key_file_path, 0, (size_t) MAX_FILE_PATH_LENGTH);
+				if (NULL == svrdb || NULL == provisioning || NULL == certificate || NULL == privateKey) {
+					THINGS_LOG_V_ERROR(THINGS_ERROR, TAG, "User certificate file not found");
+					return 0;
+				}
+
+				memset(g_svrdb_file_path, 0, (size_t) MAX_FILE_PATH_LENGTH);
+				memset(g_certificate_file_path, 0, (size_t) MAX_FILE_PATH_LENGTH);
+				memset(g_private_key_file_path, 0, (size_t) MAX_FILE_PATH_LENGTH);
+
+				if (strncmp(svrdb->valuestring, PATH_MNT, sizeof(PATH_MNT)) == 0) {
 					memcpy(g_svrdb_file_path, svrdb->valuestring, strlen(svrdb->valuestring) + 1);
-					memcpy(g_certificate_file_path, certificate->valuestring, strlen(certificate->valuestring) + 1);
-					memcpy(g_private_key_file_path, privateKey->valuestring, strlen(privateKey->valuestring) + 1);
+				} else {
+					char *svrdb_path = (char *)things_malloc(sizeof(char) * MAX_FILE_PATH_LENGTH);
+
+					strcpy(svrdb_path, PATH_MNT);
+					strcat(svrdb_path, svrdb->valuestring);
+
+					memcpy(g_svrdb_file_path, svrdb_path, strlen(svrdb_path) + 1);
+
+					things_free(svrdb_path);
+				}
+
+				if (strncmp((provisioning->valuestring), PATH_MNT, sizeof(PATH_MNT)) == 0) {
 					memcpy(g_things_cloud_file_path, provisioning->valuestring, strlen(provisioning->valuestring) + 1);
+				} else {
+					char *provisioning_path = (char *)things_malloc(sizeof(char) * MAX_CLOUD_ADDRESS);
```

Comment:
You can put string to global variable directly. There is no checking results or error case. Could you let me know the reason why you use temporary buffer through dynamic allocation?

Reply:
We have confirmed that dynamic allocation is not required.
modify